### PR TITLE
Fixes static analyzer issues from xcode.

### PR DESCRIPTION
**Pull request by sergiocampama**

### CODE REVIEW

**Comment by sergiocampama** [comment]
Fixes https://github.com/google/protobuf/issues/2034

**Comment by sergiocampama** [comment]
@pherl  FYI

**Comment by sergiocampama** [comment]
@xfxyjwf

**Comment by xfxyjwf** [comment]
I don't think these are the right changes to make. For example,
```cpp
GOOGLE_CHECK(pointer != NULL);
if (pointer != NULL) {
  ..
}
```
it seems silly to add useless checks to please a code analyzer.

**Comment by sergiocampama** [comment]
yeah, that case is what I'm trying to find a solution for...  I could replace it with assert... 

The fact that GOOGLE_CHECK is based off of the Logger is the weird part, otherwise I could add __attribute__((noreturn)) and have the analyzer understand GOOGLE_CHECK...

**Comment by xfxyjwf** [comment]
If xcode can understand some annotations, adding annotations is a more acceptable solution. Otherwise people seeing the code in protobuf internal code base may attempt to remove the additional checks you are adding here. That will be a waste of engineering resources.

**Comment by sergiocampama** [comment]
As I was telling @pherl I don't think adding the annotation to GOOGLE_CHECK is feasible, as it is done inside the logger... so we could add the noreturn attribute there, but it would be cheating, as it would return in the majority of the use cases... Another option would be to refactor GOOGLE_CHECK into something different to use an assert, for example

**Comment by sergiocampama** [comment]
Ok, removed the method one for now, what about the other ones?

**Comment by sergiocampama** [comment]
ping @xfxyjwf

File: src/google/protobuf/util/message_differencer.cc [hunk 1]
```diff
@@ -875,6 +877,7 @@ bool MessageDifferencer::CompareRepeatedField(
 
   for (int i = 0; i < count1; ++i) {
     if (match_list1[i] != -1) continue;
+    if (reporter_ == NULL) continue;
```
Comment (xfxyjwf) [inline src/google/protobuf/util/message_differencer.cc]: Make this an assert as well?

Comment (sergiocampama) [inline src/google/protobuf/util/message_differencer.cc]: Done

**Comment by xfxyjwf** [comment]
Thanks. Can you squash the commits into one? I'll merge after that.

**Comment by sergiocampama** [comment]
done... thanks! I'll try to find another way for the missing one

**Comment by sergiocampama** [comment]
the appveyor error seems to be a timeout

**Comment by xfxyjwf** [comment]
It seems appveyor is timing out on downloading dependencies very often recently...

**Pull request merged**